proto/v1/sheet_service.proto

```diff
@@ -0,0 +1,205 @@
+  // Format: projects/{project}/sheets/{sheet}
+  Sheet sheet = 1 [(google.api.field_behavior) = REQUIRED];
+
+  // The list of fields to be updated.
+  // Fields are specified relative to the sheet.
+  // (e.g. `title`, `statement`; *not* `sheet.title` or `sheet.statement`) 
+  google.protobuf.FieldMask update_mask = 2;
+}
+
+message DeleteSheetRequest {
+  // The name of the sheet to delete.
+  // Format: projects/{project}/sheets/{sheet}
+  string name = 1 [(google.api.field_behavior) = REQUIRED];
+}
+
+message SearchSheetsRequest {
+  // The parent resource of the sheet.
+  // Foramt: projects/{project}
+  string parent = 1 [(google.api.field_behavior) = REQUIRED];
+
+  // Not used. The maximum number of sheets to return. The service may return fewer than
+  // this value.
+  // If unspecified, at most 50 sheets will be returned.
+  // The maximum value is 1000; values above 1000 will be coerced to 1000.
+  int32 page_size = 2;
+
+  // Not used. A page token, received from a previous `SearchSheets` call.
+  // Provide this to retrieve the subsequent page.
+  //
+  // When paginating, all other parameters provided to `SearchSheets` must match
+  // the call that provided the page token.
+  string page_token = 3;
+}
+
+message SearchSheetsResponse {
+  // The sheets that matched the search criteria.
+  repeated Sheet sheets = 1;
+
+  // To filter the search result.
+  // Format: only support the following spec for now:
+  // - `creator == user:{email}`, `creator != user:{email}`
+  // - `starred = true`, `starred = false`.
+  string filter = 2;
+
+  // Not used. A token, which can be sent as `page_token` to retrieve the next page.
+  // If this field is omitted, there are no subsequent pages.
+  string next_page_token = 3;
+}
+
+message SyncSheetsRequest {
+  // The name of the project to sync sheets.
+  //
+  // Format: projects/{project}
+  string parent = 1;
+}
+
+message Sheet {
+  // The name of the sheet resource, generated by the server.
+  // Canonical parent is project.
+  // Format: projects/{project}/sheets/{sheet}
+  string name = 1 [(google.api.field_behavior) = REQUIRED, (google.api.field_behavior) = IMMUTABLE];
+
+  // The database resource name.
+  // Format: environments/{environment}/instances/{instance}/databases/{database}
+  // If the database parent doesn't exist, the database field is empty.
+  string database = 2;
+
+  // The title of the sheet.
+  string title = 3 [(google.api.field_behavior) = REQUIRED];
+
+  // The creator of the Sheet.
+  // Format: user:{email}
+  string creator = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // The create time of the sheet.
+  google.protobuf.Timestamp create_time = 5 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // The last update time of the sheet.
+  google.protobuf.Timestamp update_time = 6 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  string statement = 7 [(google.api.field_behavior) = REQUIRED];
```

Comment:
Also comment that the content will be cutoff and content_length below shows the full size. If they don't match, they need to call Get() with raw bool flag on.